sklearn/feature_extraction/text.py

```diff
@@ -473,6 +473,10 @@ def transform(self, X, y=None):
             Document-term matrix.
 
         """
+        if isinstance(X, str):
+            raise ValueError(
+                "iterable over raw text documents expected, string object received")
```

Comment:
Nevermind, I could see people erroneously passing in documents as strings; that would be quite annoying in console output.